apps/web/src/pages/templates/hooks/useTemplatesStoreModal.tsx

```diff
@@ -0,0 +1,21 @@
+import { useDisclosure } from '@mantine/hooks';
+
+import { useInlineComponent } from '../../../hooks';
+import { ITemplatesStoreModalProps, TemplatesStoreModal } from '../components/templates-store';
+
+const NULL_COMPONENT = () => null;
+
+export const useTemplatesStoreModal = () => {
+  const [isOpened, { open, close }] = useDisclosure(true);
+
+  const Component = useInlineComponent<ITemplatesStoreModalProps>(TemplatesStoreModal, {
+    isOpened,
+    onClose: close,
+  });
+
+  return {
+    TemplatesStoreModal: isOpened ? Component : NULL_COMPONENT,
+    openModal: open,
+    closeModal: close,
+  };
+};
```

Comment:
this is how the modal hook looks, isn't that simple and clean? :P

Reply:
It does!  💫
I will need to play with it a bit :)